src/MakingSense.AspNetCore.Documentation/DocumentationMiddleware.cs

```diff
@@ -147,14 +163,32 @@ private string GetFilePath(string subpath, out bool isDirectory)
 				: subpath;
 		}
 
-		private bool TryGetDocumentsSubpath(HttpRequest request, out PathString subpath)
+		private bool TryGetDocumentsSubpathAndLanguage(HttpRequest request, out PathString subpath, out string lang)
 		{
+			lang = null;
+
 			if (!request.Path.StartsWithSegments(_options.RequestPath, out subpath))
 			{
 				return false;
 			}
 
-			subpath = _options.FileProviderSubPath.Add(subpath);
+			string langPattern = @"^(\/[a-zA-Z]{2})(\/.*){0,1}$"; //eg. /en or /en/somedir/somefile
```

Comment:
What do you think about this expression `^\/([a-zA-Z]{2})(\/(.*)){0,1}$` ?

In that way yo do not need to trim later because the slay is outside _Group 1_:

![imagen](https://user-images.githubusercontent.com/1157864/33333503-c265fcd2-d445-11e7-9ff2-6ffa2cdcf4fc.png)

Another equivalent and more compatible alternative is without _restricted_ quantifiers (a basic state machine cannot count): `^\/([a-zA-Z][a-zA-Z])(\/(.*))?$`.

Maybe we can even use a catch-all regular expression: `^(\/([a-zA-Z][a-zA-Z]))?(\/(.*))?$`:

```
/
    Group 3.    `/`
    Group 4.    ``

/e
    Group 3.    `/e`
    Group 4.    `e`

/e/
    Group 3.    `/e/`
    Group 4.    `e/`

/es
    Group 1.    `/es`
    Group 2.    `es`
    
/es/
    Group 1.    `/es`
    Group 2.    `es`
    Group 3.    `/`
    Group 4.    ``

/es/aaaaa/bbbb/
    Group 1.    `/es`
    Group 2.    `es`
    Group 3.    `/aaaaa/bbbb/`
    Group 4.    `aaaaa/bbbb/`

/aaaaa/bbbb/
    Group 3.    `/aaaaa/bbbb/`
    Group 4.    `aaaaa/bbbb/`

/aaaaa/bbbb
    Group 3.    `/aaaaa/bbbb`
    Group 4.    `aaaaa/bbbb`
```

In that way, language will be always _Group 2_ and resource URL will be always _Group 4_